canonical/canonicalwebteam.flask-base
# [WIP] Flask 3 migration and request logs

## Done
* Upgrading to flask 3.0.0.
* Removed unsupported modules, including Talisker.
* Working on replicating required Talisker logs.


## QA
* Clone and run [dqlite.io](https://github.com/canonical/dqlite.io) locally
* Modify `requirements.txt` by commenting out `canonicalwebteam.flask-base==1.1.0` and replacing it with 
```
git+https://github.com/britneywwc/canonicalwebteam.flask-base.git@request-logs#egg=canonicalwebteam.flask-base
```
* Run the webapp with `dotrun`
* Pages should load fine and basic output logs are shown


## To Do
- [ ] Improve `http request` logs as seen in Talisker logs
- [ ] Add colour formatting for logs when in debug/local mode
